src/post.ts

```diff
@@ -1,20 +1,25 @@
 import {EventPayload} from './EventPayload';
+import useBeacon from './browser';
 
 /**
- * Used by the AnalyticsEventReport report() method to send an analytics event request via the
- * Beacon API.
- *
- * todo: Update this method to default to using fetch + keepalive unless not supported and
- * forceFetch is not set to true. Also update the description above.
+ * Used by the AnalyticsEventReport report() method to send an analytics event request via
+ * fetch, with the keepalive option set to true. If forceFetch is false and keepalive is not
+ * supported by the browser i.e. Firefox, the request is sent via the Beacon API.
  *
  * @param url URL that the request will be sent to
  * @param body the EventPayload object
+ * @param forceFetch if true, it will force the browser to call fetch instead of sendBeacon,
+ * regardless of whether keepAlive is supported
  */
-const post = (
+const post = async (
     url: string,
-    body: EventPayload
+    body: EventPayload,
+    forceFetch: boolean,
 ) => {
-    return navigator.sendBeacon(url, JSON.stringify(body));
+    if (!forceFetch && useBeacon(body)) {
+        return navigator.sendBeacon(url, JSON.stringify(body));
+    }
+    return fetch(url, {method: 'POST', body: JSON.stringify(body), keepalive: true});
 }
 
 export default post;
```

Comment:
Just noticed this when working on implementing `report`. Is there a reason to export it the function at the bottom vs doing `export post = async (` ? I noticed that when going to import the method you do it like: 

`import post from "./post";`
instead of 
`import { merge } from './merge';`

I'm sure it does not make a big difference, but figured I'd check for consistency sake

Reply:
I think that when you export a default function (which means you're only exporting a single function from the file/module), you can import it like `import post from "./post"`. You can name the method however you like when you import it e.g. `import fetch from "./post"`, `import beacon from "./post"` would both import the same method. 

When you export named exports you can export multiple in one file/module, and you will have to import it like `import { merge, helper1, helper2 } from './merge` and the names need to match how they're defined in the file. However in your case I think you can just export a default function since you only have 1 function. I might change my PR to include both functions in 1 file.